app/controllers/lead_providers/confirm_schools_controller.rb

```diff
@@ -24,20 +24,22 @@ def start
         source: :csv,
         school_ids: School.order(Arel.sql("RANDOM()")).limit(10).pluck(:id),
         delivery_partner_id: DeliveryPartner.order(Arel.sql("RANDOM()")).first.id,
-        cohort: 2021,
+        cohort_id: Cohort.current.id,
+        lead_provider_id: current_user.lead_provider.id,
       }
       redirect_to action: :show
     end
 
-    # TODO: This is temporary behaviour and will be replaced with Partnership creation
     def update
-      set_success_message heading: "The list of schools has been confirmed", content: "You will be redirected to success page in another story"
-      redirect_to action: :show
+      @confirm_schools_form.save!
+
+      redirect_to success_lead_providers_report_schools_path
```

Comment:
What happens to the session after submission? Should it be deleted?

Reply:
@kicferk1 It is being deleted on the success page